datascience/tables.py

```diff
@@ -395,6 +395,13 @@ def append_column(self, label, values):
         self._columns[label] = values
         return self
 
+    def with_column(self, label, values):
+        """Returns a new table with new column included."""
```

Comment:
Also, what does it add above `append_column`?  If this is just to make a copy, we should rather implement ``table.copy()``.